### PR TITLE
Send arguments to theano.function through kwargs argument in ComputationGraph.get_theano_function()

### PR DESCRIPTION
**Pull request by galv**
Continuation of this thread: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/blocks-users/2RWEilszr0g

It is useful in debugging to be able to send various arguments to theano.function, like MonitorMode.

Something similar could be done in GradientDescent.initialize() in order to debug the gradient descent step, but I am not sure if this is a good idea because I am not familiar with the interface.

### CODE REVIEW

File: blocks/graph.py [hunk 1]
```diff
@@ -244,12 +244,20 @@ def replace(self, replacements):
...
         return ComputationGraph(outputs_cur)
 
-    def get_theano_function(self, additional_updates=None):
-        """Create Theano function from the graph contained."""
+    def get_theano_function(self, additional_updates=None, **kwargs):
+        """Create Theano function from the graph contained.
+
+        Parameters
+        ----------
+        kwargs : key word arguments to theano.function
```
Comment (rizar) [inline blocks/graph.py]: Please see https://github.com/mila-udem/blocks/blob/master/blocks/bricks/sequence_generators.py#L814 for the proper way to add `kwargs` to `Parameters`.

File: blocks/graph.py [hunk 2]
```diff
@@ -244,12 +244,20 @@ def replace(self, replacements):
...
+            Useful for specifying specific compilation modes or profiling.
+        """
```
Comment (rizar) [inline blocks/graph.py]: Scrutinizer complains for the lack of new line in the end of the docstring.

**Comment by rizar** [comment]
Apart from the docstring looks good.

File: blocks/graph.py [hunk 3]
```diff
@@ -244,12 +244,22 @@ def replace(self, replacements):
 
         return ComputationGraph(outputs_cur)
 
-    def get_theano_function(self, additional_updates=None):
-        """Create Theano function from the graph contained."""
+    def get_theano_function(self, additional_updates=None, **kwargs):
+        """Create Theano function from the graph contained.
```
Comment (rizar) [inline blocks/graph.py]: You have to prepend the docstring with 'r' here. Not sure if I really understand why, but this is what I was advised to do and it works.

File: blocks/graph.py [hunk 4]
```diff
@@ -244,12 +244,22 @@ def replace(self, replacements):
...
+
+        Parameters
+        ----------
+        \*\*kwargs : dict
+            key word arguments to theano.function.
+            Useful for specifying specific compilation modes or profiling.
+
+        """
+
```
Comment (rizar) [inline blocks/graph.py]: Scrutinizer complains about this blank line.

**Comment by rizar** [comment]
Please rebase this PR and it will be merged. Also, I do not get information about new commits unless you write a comment.

**Comment by galv** [comment]
Sorry. I wasn't aware of that. Updates merged.

**Comment by rizar** [comment]
You have merged the recent changes, but this is not exactly what I asked you. Please follow the instruction at https://github.com/mila-udem/blocks/blob/master/docs/development/pull_request.rst#rebasing to **rebase** the PR.

**Comment by rizar** [comment]
Will this PR be rebased or should I close it?

**Comment by galv** [comment]
Hey, really sorry I totally forgot about this. Is this right?

Builds are failing, but it looks like something on your side. The place you are trying to download MNIST from is not there?

**Comment by rizar** [comment]
Thanks, now it is OK.

Tests are failing because of #836

**Pull request merged**